modules/gateway/gateway.go

```diff
@@ -159,18 +144,26 @@ func New(addr string, persistDir string) (g *Gateway, err error) {
 	g.log.Println("INFO: gateway created, started logging")
 
 	// Forward the RPC port, if possible.
-	go g.forwardPort(port)
+	g.threads.Add()
+	go func() {
+		defer g.threads.Done()
+		g.forwardPort(port)
```

Comment:
However, it is `g.forwardPort`, and there's a pretty strict convention that you need to assume unexported methods on an object hold a lock. So, either you need to change it to `g.threadedForwardPort`, `g.ForwardPort`, or `forwardPort` (last one is not a method on the gateway).

Definitely the way it's being used here is incorrect, you can't call an unexported method on the gateway in a goroutine unless said goroutine wraps the call in a lock.

Given that no state is changing, I'd probably prefer `forwardPort`